src/angular/components/sso.component.ts

```diff
@@ -66,34 +77,42 @@ export class SsoComponent {
         });
     }
 
-    async submit() {
-        const passwordOptions: any = {
-            type: 'password',
-            length: 64,
-            uppercase: true,
-            lowercase: true,
-            numbers: true,
-            special: false,
-        };
+    async buildAuthorizeUrl(): Promise<string> {
         let codeChallenge = this.codeChallenge;
         let state = this.state;
         if (codeChallenge == null) {
-            const codeVerifier = await this.passwordGenerationService.generatePassword(passwordOptions);
+            const codeVerifier = await this.passwordGenerationService.generatePassword(this.passwordOptions);
             const codeVerifierHash = await this.cryptoFunctionService.hash(codeVerifier, 'sha256');
             codeChallenge = Utils.fromBufferToUrlB64(codeVerifierHash);
             await this.storageService.save(ConstantsService.ssoCodeVerifierKey, codeVerifier);
         }
+
         if (state == null) {
-            state = await this.passwordGenerationService.generatePassword(passwordOptions);
+            state = await this.passwordGenerationService.generatePassword(this.passwordOptions);
+            if (this.returnUri) {
+                state += `_returnUri='${this.returnUri}'`;
+            }
+
             await this.storageService.save(ConstantsService.ssoStateKey, state);
         }
 
-        const authorizeUrl = this.apiService.identityBaseUrl + '/connect/authorize?' +
+        let authorizeUrl = this.apiService.identityBaseUrl + '/connect/authorize?' +
             'client_id=' + this.clientId + '&redirect_uri=' + encodeURIComponent(this.redirectUri) + '&' +
             'response_type=code&scope=api offline_access&' +
             'state=' + state + '&code_challenge=' + codeChallenge + '&' +
             'code_challenge_method=S256&response_mode=query&' +
             'domain_hint=' + encodeURIComponent(this.identifier);
+
+        if (this.linkLoggedInUser) {
```

Comment:
Maybe change this to a function param of `includeUserIdentifier` or something. Default it to false.

Reply:
Done. Instead of defaulting to false I just made the parameter optional, but the effect is the same.